rust_filter/src/graph_utils.rs

```diff
@@ -92,12 +96,36 @@ pub fn generate_trace_graph_from_headers(
             prev_node_handle = node_str_to_node_handle[node_str];
         }
     }
+
+    // now insert properties:
+    // do we have any?
+    let comma_index = properties_header.find(",");
+    if !comma_index.is_none() {
+        let properties_iterator = properties_header.split(",");
+        for statement in properties_iterator {
+            if !statement.find("==").is_none() {
```

Comment:
What does this check for?

Reply:
It's checking if there are any properties at all.  If the node is the first one visited, the properties_header will have nothing in it, so there won't be a comma after a property.

Reply:
What if you just use the split to check? `let mut values_iterator = statement.split("==");` Same for the commas

Reply:
If you split by something not in the string, then you still get one iteration from .split().  So if I first split

a.envoy.property==value,

by a comma, then I wind up with two chunks that the iterator goes over, one containing "a.envoy.property==value" and the other just as "".  So I always need to be checking for the last empty string, which I why I'm still checking that the "==" even exists after I've done the comma split.